### PR TITLE
Pass rep into GetEntryFromCache function.

### PR DESCRIPTION
**Pull request by HaoyuHuang**
We plan to log block cache access in GetEntryFromCache. Passing rep object so that we can access fd number, column family information in GetEntryFromCache.

### CODE REVIEW

File: table/block_based_table_reader.cc [hunk 1]
```diff
@@ -129,18 +129,17 @@ void ForceReleaseCachedEntry(void* arg, void* h) {
...
-                                 uint64_t* block_cache_miss_stats,
-                                 uint64_t* block_cache_hit_stats,
-                                 Statistics* statistics,
-                                 GetContext* get_context) {
+Cache::Handle* GetEntryFromCache(
+    Cache* block_cache, const Slice& key, const BlockBasedTable::Rep* rep,
+    Tickers block_cache_miss_ticker, Tickers block_cache_hit_ticker,
+    uint64_t* block_cache_miss_stats, uint64_t* block_cache_hit_stats,
+    Statistics* statistics, GetContext* get_context) {
+  assert(rep);
```
Comment (siying) [inline table/block_based_table_reader.cc]: If we even need to pass `rep` to this function, is there any reason not to make it a function of class `BlockBasedTable`?

Comment (HaoyuHuang) [reply table/block_based_table_reader.cc]: No. The class has some static functions. I'm doing a major refactoring to remove them right now. I will send out a new PR for it.

Comment (HaoyuHuang) [reply table/block_based_table_reader.cc]: Thanks for your comment. I refactored the code to make GetEntryFromCache to become a member function of BlockBasedTable.

**Comment by riversand963** [comment]
I noticed that a non-negligible portion of this PR results from `make format`. Although not official, it is usually better to void combining irrelevant format change with your real code change.

**Pull request closed without merging**